tests/testthat/test-misty.R

```diff
@@ -79,3 +79,27 @@ test_that("run_misty handles tests of failures",{
   expect_true(any(grepl("R2", sig.warnings)))
   unlink("results", recursive = TRUE)
 })
+
+test_that("modeling of intraview is bypassed if only 1 var in intraview", {
+  truncated_expr <- generate_random_tibble(100, 1)
+  pos <- sample_grid_geometry(100, 10, 10)
+  misty.views <- create_initial_view(truncated_expr)
+  sig.warnings <- capture_warnings(misty.views %>% run_misty())
+  expect_true(any(grepl("Negative performance", sig.warnings)))
+})
+
+test_that("warning raised if variance of target is 0", {
+  expr <- tibble::tibble(expr1 = rnorm(100, 10, 2),
+                         expr2 = rep(10, 100))
+  pos <- sample_grid_geometry(100, 10, 10)
+  misty.views <- create_initial_view(expr)
+  sig.warnings <- capture_warnings(misty.views %>% run_misty())
+  expect_true(any(grepl("Targets expr2 have zero variance", sig.warnings)))
```

Comment:
It is good but just a thought. Can this be made more general somehow? What if we change the warning message?

Reply:
testthat::expect_warning() only?

Reply:
But the code results in several warnings, how can we use expect_warning() in that case?

Reply:
better toy dataset, where the only expected warning to be raised is the zero variance one.